### PR TITLE
Np 3598 random scopus generator

### PR DESCRIPTION
**Pull request by axthosarouris**
First iteration of a scopus data generator.

### CODE REVIEW

File: create-publication-from-scopus/src/main/resources/conversion/ignoredScopusFields.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.meta
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/main/resources/conversion/ignoredScopusFields.txt:1]: This file will contain all fields that we know we are ignoring. It has the ability to list nested files as well.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/test/utils/ScopusGenerator.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public static final Set<Class<?>> NOT_BEAN_CLASSES = Set.of(XMLGregorianCalendar.class);
+    private static final Set<String> IGNORED_FIELDS = readIgnoredFields();
+
+    public ScopusGenerator() {
+
+    }
+
+    public static DocTp randomDocument() {
+        DocTp docTp = new DocTp();
+        assertThat(docTp, doesNotHaveEmptyValuesIgnoringFieldsAndClasses(NOT_BEAN_CLASSES, IGNORED_FIELDS));
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/test/utils/ScopusGenerator.java:23]: currently everything is ignored. More functionality comes soon.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -37,7 +36,6 @@ nva-doi = { group = "com.github.bibsysdev", name = "doi", version.ref = "nva" }
 nva-testutils = { group = "com.github.bibsysdev", name = "nvatestutils", version.ref = "nva" }
 nva-datamodel = { group = "com.github.bibsysdev", name = "nva-datamodel-java", version.ref = "nvaDatamodelVersion" }
 
-testUtils = { group = "com.github.BIBSYSDEV", name = "nva-testutils", version.ref = "testUtils" }
```
Comment (axthosarouris) [inline gradle/libs.versions.toml:40]: this is an obsolete package.

**Review by anetteOlli — APPROVED** [review]

**Review by nraanes — APPROVED** [review]

File: create-publication-from-scopus/src/main/resources/conversion/ignoredScopusFields.txt (hunk 1 above)
Comment (gregorgabriel) [reply create-publication-from-scopus/src/main/resources/conversion/ignoredScopusFields.txt:1]: Riktig! Ignore Meta. Siden den er Facebook sin.

**Review by gregorgabriel — CHANGES_REQUESTED** [review]
I still dont see how this should help us testing Scopus import. Do you have to have a new ignoredScopusFields for each testcase? Or do you will have the possibility to open up specific ignored fields in the testscope?
I would like to see some example (pseudocode will do).

**Comment by axthosarouris** [comment]
> I still dont see how this should help us testing Scopus import. Do you have to have a new ignoredScopusFields for each testcase? Or do you will have the possibility to open up specific ignored fields in the testscope? I would like to see some example (pseudocode will do).

Example: https://github.com/BIBSYSDEV/nva-fetch-doi/pull/245/files

**Review by brinxmat — APPROVED** [review]

**Review by gregorgabriel — COMMENTED** [review]
I abstain my vote. See comment on next PR on that subject.

**Review by gregorgabriel — APPROVED** [review]

**Pull request merged**